src/test/java/com/google/devtools/build/lib/actions/ActionCacheCheckerTest.java

```diff
@@ -368,15 +412,12 @@ public void testMiddleman_differentFiles() throws Exception {
   public void testDeletedConstantMetadataOutputCausesReexecution() throws Exception {
     SpecialArtifact output =
         new Artifact.SpecialArtifact(
-            ArtifactRoot.asDerivedRoot(
-                new InMemoryFileSystem(DigestHashFunction.SHA256).getPath("/output"),
-                RootType.Output,
-                "bin"),
+            artifactRoot,
             PathFragment.create("bin/dummy"),
-            ActionsTestUtil.NULL_ARTIFACT_OWNER,
+            NULL_ARTIFACT_OWNER,
             SpecialArtifactType.CONSTANT_METADATA);
     output.getPath().getParentDirectory().createDirectoryAndParents();
-    Action action = new NullAction(output);
+    Action action = new WriteEmptyOutputAction(output);
```

Comment:
Previously, outputs are [created](https://github.com/bazelbuild/bazel/blob/2c059126de8e128d9dc1cc8b7eb3d7fe99bde6e6/src/test/java/com/google/devtools/build/lib/actions/ActionCacheCheckerTest.java#L123) with empty content within `runAction`. Since we need to test against the case that outputs are not created but instead metadata are  injected, we move the creation to the action itself.